internal/dependency/registry/version.go

```diff
@@ -22,7 +22,7 @@ func (ver Version) GreaterThan(ver2 Version) bool {
 
 func getExactMatch(version Version) (string, error) {
 	_, err := semver.NewVersion(string(version))
-	if err != nil {
+	if err != nil && version != "*" {
 		return "", err
```

Comment:
don't you want to return a non-nil error in case if you got here because of the 2nd part of the condition?

Reply:
I don't think I follow. This is `and` condition. You get there because of both parts.

Reply:
Ok, right. I knew something was off with this complex condition, but you're right - that's not it.
What I should've said was: what if err is not nil and the version is "*"? (but I guess the case is so specific that you've already made sure it's what you actually want)